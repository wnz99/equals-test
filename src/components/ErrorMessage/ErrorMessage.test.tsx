import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import React from 'react'

import { ErrorMessage } from './index'

const mockOnRetry = jest.fn()

describe('<ErrorMessage />', () => {
  beforeEach(() => {
    mockOnRetry.mockClear()
  })

  it('should render without errors', async () => {
    const { asFragment } = render(<ErrorMessage />)

    expect(asFragment()).toMatchSnapshot()

    expect(screen.getByText(/There has been an error./i)).toBeInTheDocument()
  })

  it('should show a custom error message', async () => {
    render(<ErrorMessage message="custom message" />)

    expect(screen.getByText(/custom message/i)).toBeInTheDocument()
  })

  it('should trigger callback when user click Add button', async () => {
    render(<ErrorMessage onRetry={mockOnRetry} />)

    await waitFor(async () => {
      fireEvent.click(screen.getByText(/Retry/i))

      expect(mockOnRetry).toHaveBeenCalledTimes(1)
    })
  })
})
