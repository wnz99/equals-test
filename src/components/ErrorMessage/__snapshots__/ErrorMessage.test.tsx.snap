// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> should render without errors 1`] = `
<DocumentFragment>
  <div
    class="css-eejnqe"
  >
    <div
      class="css-rwk5vx"
    >
      <p
        class="chakra-text css-zwkqrq"
      >
        There has been an error.
      </p>
      <button
        class="chakra-button css-4xx2wk"
        type="button"
      >
        Retry
      </button>
    </div>
  </div>
</DocumentFragment>
`;
